### ISSUE TITLE
use randomly generate source code to test decompyle3

### ISSUE THREAD

**Issue by 15r10nk** [rocky/python-decompile3#170]
## Description

Hi, thank you for this very cool tool. 
I wrote [pysource-codegen](https://github.com/15r10nk/pysource-codegen) to to test some of the software I wrote.
I would recommend you to use it in combination with [pysource-minimize](https://github.com/15r10nk/pysource-minimize) to generate minimal reproductions for the bugs.

Let me know if you are interested or if I can help you with the integration.

**Comment by rocky** [comment]
Thanks for the kind words.

I am currently on my way to [BlackHat Asia 2024 where I'll be speaking on decomple3 ](https://www.blackhat.com/asia-24/briefings/schedule/index.html#how-to-get-the-most-out-of-the-python-decompilers-uncompyle-and-decompyle---how-to-write-and-read-a-bytecode-decompiler-37789)

As it says in [here](https://github.com/rocky/python-decompile3/blob/master/HOW-TO-REPORT-A-BUG.md#the-importance-of-your-bug-report): 

> We already have an infinite supply of decompilation bugs that need fixing, and an automated mechanism for finding more.

There is however a bit of time spent in _narrowing_ code to a minimum program that fails to decompile. So it is possible that something along the lines of pysource-minimize can be useful. I will look into this more when I get back in a month or so. (I will take a vacation afterwards).  

I suspect though that a tool for minimizing a problem will be more tied to understanding more about specifics of bytecode. For example, the code doesn't have to run, just be compilable which means syntactically valid. For example, undefined variables or referring to attributes of an object that does not exist usually okay. Having dead code is okay. None of these generally cause byte compiling to fail.

**Comment by 15r10nk** [comment]
Yes, it is a good idea to start with the problems you already have.

I think the minimization should work for you, because it does exactly what you says. It minimizes the code as long as it can be compiled and reproduce your bug. It tries not to be  smart about unused code or variables.

I hope this will save you some time in understanding and fixing your bugs.
Let me know about your experience and have a nice vacation.
